src/DataWrangling/ecco_restoring.jl

```diff
@@ -0,0 +1,284 @@
+- time_indices_in_memory: The number of time indices to keep in memory (default: 2).
+- time_indexing: The time indexing scheme to use (default: Cyclical()).
+- grid: if not a `nothing`, the ECCO data is directly interpolated on the `grid`,
+"""
+function ECCO_field_time_series(metadata::ECCOMetadata;	
+                                architecture = CPU(),	
+                                time_indices_in_memory = 2,	
+                                time_indexing = Cyclical(),
+                                grid = nothing)	
+
+    # ECCO data is too chunky to allow other backends	
+    backend = ECCONetCDFBackend(time_indices_in_memory; 
+                                on_native_grid = isnothing(grid))
+
+    # Making sure all the required individual files are downloaded
+    download_dataset!(metadata)
+
+    location = field_location(metadata)
+    ftmp = empty_ecco_field(first(metadata); architecture)
+    shortname = short_name(metadata)
+
+    ECCO_native_grid = ftmp.grid
+    boundary_conditions = FieldBoundaryConditions(ECCO_native_grid, location)
+    times = ecco_times(metadata)
+
+    fts_grid = isnothing(grid) ? ECCO_native_grid : grid
+
+    fts = FieldTimeSeries{location...}(fts_grid, times;	
+                                       backend,	
+                                       time_indexing,	
+                                       boundary_conditions,	
+                                       path = metadata,	
+                                       name = shortname)
+
+    # Let's set the data	
+    set!(fts)	
+
+    return fts	
+end
+
+ECCO_field_time_series(variable_name::Symbol, version=ECCO4Monthly(); kw...) = 
+    ECCO_field_time_series(ECCOMetadata(variable_name, all_ecco_dates(version), version); kw...)
+
+# Variable names for restoreable data
+struct Temperature end
+struct Salinity end
+struct UVelocity end
+struct VVelocity end
+
+oceananigans_fieldname = Dict(
+    :temperature => Temperature(), 
+    :salinity    => Salinity(), 
+    :u_velocity  => UVelocity(), 
+    :v_velocity  => VVelocity())
+
+@inline Base.getindex(fields, i, j, k, ::Temperature) = @inbounds fields.T[i, j, k]
+@inline Base.getindex(fields, i, j, k, ::Salinity)    = @inbounds fields.S[i, j, k]
+@inline Base.getindex(fields, i, j, k, ::UVelocity)   = @inbounds fields.u[i, j, k]
+@inline Base.getindex(fields, i, j, k, ::VVelocity)   = @inbounds fields.v[i, j, k]
+
+"""
+    struct ECCORestoring{FTS, G, M, V, N} <: Function
+
+A struct representing ECCO restoring.
+
+# Fields
+- `ecco_fts`: The ECCO FTS on the native ECCO grid.
+- `ecco_grid`: The native ECCO grid to interpolate from.
+- `mask`: A mask (could be a number, an array, a function or a field).
+- `varname`: The variable name of the variable that needs restoring.
+- `λ⁻¹`: The reciprocal of the restoring timescale.
+"""
+struct ECCORestoring{FTS, G, M, V, N} <: Function
+    ecco_fts  :: FTS
+    ecco_grid :: G
+    mask      :: M
+    varname   :: V
```

Comment:
in the JRA55 module we use `variable_name`. Do we want to use `varname` or `variable_name`? We should pick one.

Reply:
I prefer `varname` (personal preference) but I am ok with `variable_name`.

Reply:
It just departs from the rest of the style of the code which is to use words for names

Reply:
ok, I have changed it to variable_name